tests/test_diag_coded_types.py

```diff
@@ -694,9 +703,10 @@ def test_encode_min_max_length_type_zero(self) -> None:
             termination="ZERO",
             is_highlow_byte_order_raw=None,
         )
-        state = EncodeState(bytearray([0x12]), parameter_values={}, is_end_of_pdu=False)
-        byte_val = dct.convert_internal_to_bytes("Hi", state, bit_position=0)
-        self.assertEqual(byte_val, bytes([0x48, 0x69, 0x0]))
+        state = EncodeState(
+            coded_message=bytearray([0x00]), parameter_values={}, is_end_of_pdu=False)
```

Comment:
why was the 0x12 replaced with zero?

Reply:
that was done to avoid those pesky "overlapping parameter" warnings: `convert_internal_to_bytes()` simply did not care what already was in the PDU, `encode_into_pdu()` does...